openmc/data/multipole.py

```diff
@@ -129,6 +143,600 @@ def _broaden_wmp_polynomials(E, dopp, n):
     return factors
 
 
+def _vectfit_xs(energy, ce_xs, mts, rtol=1e-3, atol=1e-5, orders=None,
+                n_vf_iter=30, log=False, path_out=None):
+    """Convert point-wise cross section to multipole data via vector fitting.
+
+    Parameters
+    ----------
+    energy : np.ndarray
+        Energy array
+    ce_xs : np.ndarray
+        Point-wise cross sections to be fitted, with shape (number of reactions,
+        number of energy points)
+    mts : Iterable of int
+        Reaction list
+    rtol : float, optional
+        Relative error tolerance
+    atol : float, optional
+        Absolute error tolerance
+    orders : Iterable of int, optional
+        A list of orders (number of poles) to be searched
+    n_vf_iter : int, optional
+        Number of maximum VF iterations
+    log : bool or int, optional
+        Whether to print running logs (use int for verbosity control)
+    path_out : str, optional
+        Path to save the figures to show discrepancies between the original and
+        fitted cross sections for different reactions
+
+    Returns
+    -------
+    tuple
+        (poles, residues)
+
+    """
+
+    # import vectfit package: https://github.com/liangjg/vectfit
+    import vectfit as vf
+
+    ne = energy.size
+    nmt = len(mts)
+    if ce_xs.shape != (nmt, ne):
+        raise ValueError('Inconsistent cross section data.')
+
+    # construct test data: interpolate xs with finer grids
+    n_finer = 10
+    ne_test = (ne - 1)*n_finer + 1
+    test_energy = np.interp(np.arange(ne_test),
+                            np.arange(ne_test, step=n_finer), energy)
+    test_energy[[0, -1]] = energy[[0, -1]] # avoid numerical issue
+    test_xs_ref = np.zeros((nmt, ne_test))
+    for i in range(nmt):
+        test_xs_ref[i] = np.interp(test_energy, energy, ce_xs[i])
+
+    if log:
+        print("Energy: {:.3e} to {:.3e} eV ({} points)".format(
+              energy[0], energy[-1], ne))
+
+    # transform xs (sigma) and energy (E) to f (sigma*E) and s (sqrt(E)) to be
+    # compatible with the multipole representation
+    f = ce_xs * energy
+    s = np.sqrt(energy)
+    test_s = np.sqrt(test_energy)
+
+    # inverse weighting is used for minimizing the relative deviation instead of
+    # absolute deviation in vector fitting
+    weight = 1.0/f
```

Comment:
It looks like in some cases, we get divide by zeros (for example, He4 from ENDF/B-VIII.0). Will this cause any problems?

Reply:
After more testing, it seems like this is especially common for fissionable nuclides

Reply:
You're right. For the cross section that is too small (less than 1e-7, including zeros), the weight will be adjusted specifically, as shown in [the following lines](https://github.com/openmc-dev/openmc/blob/b842ae78eed362af216c23bfb25ae86620fec7d0/openmc/data/multipole.py#L212-L219).

Reply:
In that case, we probably ought to ignore divide by zeros:
```suggestion
    with np.errstate(divide='ignore'):
        weight = 1.0/f
```

Reply:
ok